src/main/java/io/cdap/plugin/SFTPCopyAction.java

```diff
@@ -129,65 +131,74 @@ public String getVariableNameHoldingFileList() {
   @Override
   public void run(ActionContext context) throws Exception {
     Path destination = new Path(config.getDestDirectory());
-
     Configuration conf = new Configuration();
     Map<String, String> properties = config.getFileSystemProperties();
     for (Map.Entry<String, String> entry : properties.entrySet()) {
       conf.set(entry.getKey(), entry.getValue());
     }
-
     FileSystem fileSystem = FileSystem.get(conf);
-
     destination = fileSystem.makeQualified(destination);
     if (!fileSystem.exists(destination)) {
       fileSystem.mkdirs(destination);
     }
-
-    try (SFTPConnector SFTPConnector = new SFTPConnector(config.getHost(), config.getPort(), config.getUserName(),
-                                                      config.getPassword(), config.getSSHProperties())) {
-      ChannelSftp channelSftp = SFTPConnector.getSftpChannel();
-
-      Vector files = channelSftp.ls(config.getSrcDirectory());
-
-      List<String> filesCopied = new ArrayList<>();
-      for (int index = 0; index < files.size(); index++) {
-        Object obj = files.elementAt(index);
-        if (!(obj instanceof ChannelSftp.LsEntry)) {
-          continue;
-        }
-        ChannelSftp.LsEntry entry = (ChannelSftp.LsEntry) obj;
-        if (".".equals(entry.getFilename()) || "..".equals(entry.getFilename())) {
-          // ignore "." and ".." files
-          continue;
+    SFTPConnector sftpConnector = null;
+    try {
+      if (config.getAuthTypeBeingUsed().equals(SFTPConstants.PRIVATE_KEY_SELECT)) {
+        sftpConnector = new SFTPConnector(config.getHost(), config.getPort(),
+          config.getUserName(), config.getPrivateKey(), config.getPassphrase(), config.getSSHProperties());
+      } else {
+        sftpConnector = new SFTPConnector(config.getHost(), config.getPort(),
+          config.getUserName(), config.getPassword(), config.getSSHProperties());
+      }
+      copySFTPFiles(fileSystem, destination, sftpConnector, context);
+    } catch(Exception e) {
+      throw new RuntimeException(String.format("Error occurred while connecting to SFTP Server %s", e.getMessage()));
```

Comment:
also add exception:
`new RuntimeException(String.format("Error occurred while connecting to SFTP Server %s", e.getMessage(), e));`